vignettes/shiny-liquid.Rmd

```diff
@@ -0,0 +1,311 @@
+which we can use to generate a configuration for that webpack.
+Let's just pick CSS from the Styling section and copy the the script to `js/webpack.config.js`.
+We also add dev dependencies as suggested by the tool:
+```sh
+yarn add --dev webpack webpack-cli css-loader style-loader
+```
+
+Now let's tweak the config a bit.
+We change the output to `inst/www/liquid`:
+```js
+output: {
+  path: path.join(__dirname, '..', 'inst', 'www'),
+  filename: 'liquid.js'
+}
+```
+
+We add [`externals`](https://webpack.js.org/configuration/externals/)
+to let webpack know where to look for modules provided by shiny.react:
+```js
+externals: {
+  'react': 'jsmodule["react"]',
+  'react-dom': 'jsmodule["react-dom"]',
+  '@/shiny.react': 'jsmodule["@/shiny.react"]'
+}
+```
+
+Our final `js/webpack.config.js` looks as follows:
+```js
+const webpack = require('webpack');
+const path = require('path');
+
+const config = {
+  entry: './src/index.js',
+  output: {
+    path: path.join(__dirname, '..', 'inst', 'www'),
+    filename: 'liquid.js'
+  },
+  module: {
+    rules: [
+      {
+        test: /\.css$/,
+        use: [
+          'style-loader',
+          'css-loader'
+        ]
+      }
+    ]
+  },
+  externals: {
+    'react': 'jsmodule["react"]',
+    'react-dom': 'jsmodule["react-dom"]',
+    '@/shiny.react': 'jsmodule["@/shiny.react"]'
+  }
+};
+
+module.exports = config;
+```
+
+
+### Building the package
+We are ready to build our package!
+First of all, we run webpack in the `js` directory:
+```sh
+yarn webpack
+```
+This will generate the `inst/www/webpack.js` bundle.
+We should also generate the NAMESPACE file:
+```r
+devtools::document()
+```
+
+We can now install the package directly with `devtools::install()` and try it out!
+
+### Using the package
+Let's try a simple app first to test our components:
+```r
+library(shiny)
+library(shiny.react)
+library(liquid)
+
+shinyApp(
+  ui = tagList(
+    LdButton("Test Button"),
+    LdLoading()
+  ),
+  server = function(input, output) {}
+)
+```
+
+Cool! Let's try something more advanced:
+```r
+shinyApp(
+  ui = tagList(
+    LdButton(
+      "Initiate loading",
+      onClick = JS("(event) => Shiny.setInputValue('loading', true)")
+    ),
+    reactOutput("spinner")
+  ),
+  server = function(input, output) {
+    output$spinner <- renderReact({
+      if (!is.null(input$loading) && input$loading) LdLoading() else NULL
+    })
+  }
+)
+```
+
+### Creating input wrappers
+Even simple components can be cumbersome to use in Shiny, as evident in the last example.
+It is a good idea to create `.shinyInput` wrappers to simplify the life of your users.
+
+We change our `js/src/index.js` to the following:
+```js
+const Liquid = require('@emdgroup-liquid/liquid/dist/react');
+const { InputAdapter } = require('@/shiny.react')
+
+require('@emdgroup-liquid/liquid/dist/css/liquid.css');
+
+const LdSelect = InputAdapter(Liquid.LdSelect, (value, setValue) => ({
```

Comment:
Maybe we could add some background what `InputAdapter` does and why we use `onLdchange` here?

> In order to create an input that can be used in Shiny server we need to create the component with a [hook](https://en.reactjs.org/docs/hooks-intro.html) that will set a value of Shiny input. The [documentation](https://emdgroup-liquid.github.io/liquid/introduction/event-handling/) states that Liquid components dispatch `ldchange` events, to change value of Shiny input we need to set a value when component changes its state. For React components we use `onLdchange` prop and we set the value using `event.detail` which contains an array of selected items from the dropdown. If the documentation provides information which `event` field contains value of input use the one from documentation, if it doesn't you can set a breakpoint in the browser to investigate what fields does `event` object have and use the appropriate one

Since we encountered some issues with porting this component it could be informative to include description what issues users may face. This way we may limit the number of potential issues raised by users when they try to create their own packages